src/btree/bt_sync.c

```diff
@@ -153,6 +157,9 @@ __sync_file(WT_SESSION_IMPL *session, int syncop)
 				++leaf_pages;
 			}
 			WT_ERR(__wt_reconcile(session, walk, NULL, 0));
+
+			if (TXNID_LT(btree->rec_max_txn, mod->rec_max_txn))
+				btree->rec_max_txn = mod->rec_max_txn;
```

Comment:
I'd put the common update code in one place:
```
118a119,122
>                       if (mod != NULL &&
>                           TXNID_LT(btree->rec_max_txn, mod->rec_max_txn))
>                               btree->rec_max_txn = mod->rec_max_txn;
>
120,123c124
<                       if (!__wt_page_is_modified(page)) {
<                               if (mod != NULL && TXNID_LT(
<                                   btree->rec_max_txn, mod->rec_max_txn))
<                                       btree->rec_max_txn = mod->rec_max_txn;
---
>                       if (!__wt_page_is_modified(page))
125d125
<                       }
160,162d159
<
<                       if (TXNID_LT(btree->rec_max_txn, mod->rec_max_txn))
<                               btree->rec_max_txn = mod->rec_max_txn;
```
and let the compiler do the work.

It also would add a natural place to add a comment explaining what's happening here, I think that's worth doing regardless.